test/modules/core/webgl/framebuffer.spec.js

```diff
@@ -193,36 +194,74 @@ test('WebGL1#Framebuffer resize', t => {
   t.end();
 });
 
-function testFramebufferReadPixels(t, gl) {
-  const frameBufferOptions = {
-    attachments: {
-      [GL.COLOR_ATTACHMENT0]: new Texture2D(gl),
-      [GL.DEPTH_STENCIL_ATTACHMENT]: new Renderbuffer(gl, {format: GL.DEPTH_STENCIL})
-    }
-  };
+const FB_READPIXELS_TEST_CASES = [
+  {
+    format: GL.RGBA, clearColor: [1, 0.5, 0.25, 0.125], expectedColor: [255, 128, 64, 32]
+  },
 
-  const framebuffer = new Framebuffer(gl, frameBufferOptions);
+  // TODO: Framebuffer creation fails under Node (browser WebGL1 is fine)
+  // {
+  //   format: GL.RGB, clearColor: [1, 0.25, 0.125, 0], expectedColor: [255, 64, 32]
+  // },
 
-  framebuffer.resize({width: 1000, height: 1000});
-  framebuffer.checkStatus();
+  {
+    format: GL.RGBA32F, clearColor: [0.214, -32.23, 1242, -123.847]
+  },
+  {
+    format: GL.RG32F, clearColor: [-0.214, 32.23, 0, 0], expectedColor: [-0.214, 32.23, 0, 1] // ReadPixels returns default values for un-used channels (B and A)
```

Comment:
That seems wasteful, I take it readpixels can't be set to `GL.RG`?

It could be worth updating our docs with these findings. E.g. FrameBuffer.readPixels doc, and maybe start writing users guide on textures.

Or at least start collecting notes/bullets of your discoveries.

Reply:
Added details to `texture.md` and `framebuffer.md`.